### PR TITLE
fix(Core/Pathfinding): Creatures regen health, spells LOD

### LINKED ISSUE #4436: [raid]Archimonde Can't be killed
##### CURRENT BEHAVIOUR:
Archimonde health reduced to 10% will enter the plot, and then the elf will go over and explode, Archimonde health reduced to 7%, out of combat. Archimonde health restored

##### EXPECTED BLIZZLIKE BEHAVIOUR:
Archimonde health reduced to 10% will enter the plot, and then the elf will go and explode until Archimonde dies

##### STEPS TO REPRODUCE THE PROBLEM:
1. update to https://github.com/azerothcore/azerothcore-wotlk/commit/504a788c63265a85ac730ce5dd6c5a067f25ee67
2. .go c 39927
3. Attack Archimonde health to 10%
4. Then watch the story

##### AC HASH/COMMIT:
https://github.com/azerothcore/azerothcore-wotlk/commit/504a788c63265a85ac730ce5dd6c5a067f25ee67

##### OPERATING SYSTEM:
ubuntu 18.04
##### MODULES:
none

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/96382397-raid-archimonde-can-t-be-killed?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by Footman**
## Changes Proposed:
- fix path generation for spells. sometimes tracing throw collision. For example blinking sometimes throw earth.
- fix bosses and other creatures regenerating health instantly when cannot reach target. fix #4436 #4455
- fix creatures moonwalking after moveback + death 

## Target Branch(es):
- [x] Master

 
## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

### CODE REVIEW

**Comment by Voxstrasza** [comment]
@Footman I haven't tested if it's an issue yet but would this work if Lich King potentially regenerated health during the roleplay in phase 4 (after he kills the raid etc)

**Comment by Footman** [comment]
> @Footman I haven't tested if it's an issue yet but would this work if Lich King potentially regenerated health during the roleplay in phase 4 (after he kills the raid etc)

there was no such issue as far i know because lich could stay in fight during this phase before as well.

File: src/server/game/Movement/Spline/MoveSpline.h [hunk 1]
```diff
@@ -112,6 +112,10 @@ namespace Movement
 
         bool onTransport;
         [[nodiscard]] std::string ToString() const;
+         bool HasStarted() const
+        {
+            return time_passed > 0;
+        }
```
Comment (FrancescoBorzi) [inline src/server/game/Movement/Spline/MoveSpline.h:118]: brackets not aligned also please add `[[nodiscard]]`

**Comment by FrancescoBorzi** [comment]
@mpfans said he did some testing with real players and we can merge this

**Review by FrancescoBorzi — APPROVED** [review]

**Pull request merged**